src/Functions/FunctionBase64Conversion.h

```diff
@@ -85,6 +85,85 @@ struct TryBase64Decode
     }
 };
 
+struct Base64UrlEncode : Base64Encode
```

Comment:
> Could we use templatization instead of inheritance, please?
> 
> Instead of l. 88-165 so something like this
> 
> ```c++
> enum class Base64Variant
> {
>     Normal,
>     Url
> };
> 
> void preprocessBase64Url(...)
> {
>     ...
> }
> 
> void postprocessBase64Url(...)
> {
>     ...
> }
> 
> template <Base64Variant variant>
> struct Base64Encode
> {
>     static constexpr auto name = (variant == Base64Variant::Normal) ? "base64Encode" : "base64UrlEncode";
> 
>     static size_t getBufferSize(size_t string_length, size_t string_count)
>     {
>         return ((string_length - string_count) / 3 + string_count) * 4 + string_count;
>     }
> 
>     static size_t perform(const std::span<const UInt8> src, UInt8 * dst)
>     {
>         if constexpr (variant == Base64Variant::Url)
>             preprocess(src, dst);
> 
>         size_t outlen = 0;
>         base64_encode(reinterpret_cast<const char *>(src.data()), src.size(), reinterpret_cast<char *>(dst), &outlen, 0);
>         return outlen;
>     }
> };
> ```

Yes, sure.